### ISSUE TITLE
PackedSequence's sorted_indices is not put on cuda when to('cuda') is called.

### ISSUE THREAD

**Issue by enhuiz** [pytorch/pytorch#22251]
## 🐛 Bug

PackedSequence's sorted_indices is not put on cuda when `.to('cuda')` is called, but `.to(device='cuda')` works. The problem seems to be here:

https://github.com/pytorch/pytorch/blob/c1fc2f25c2ca58cd6f39c96193f2183a2a561e7b/torch/nn/utils/rnn.py#L194

Maybe it should be `lambda t: t.to(*args, **kwargs)` as the previous line:

https://github.com/pytorch/pytorch/blob/c1fc2f25c2ca58cd6f39c96193f2183a2a561e7b/torch/nn/utils/rnn.py#L189

###  

```python
import torch
from torch.nn.utils.rnn import pack_sequence, pad_packed_sequence

seqs = [torch.randn(i, 1) for i in range(10, 0, -1)]
seqs = pack_sequence(seqs, enforce_sorted=False)
seqs = seqs.to('cuda') # .to('cuda') causes the bug, but .to(device='cuda') won't
print(seqs.sorted_indices.device)

try:
    pad_packed_sequence(seqs, True, 0)
except Exception as e:
    print(e)
```

outputs

```
cpu
Expected object of backend CUDA but got backend CPU for argument #3 'index'
```

## Expected behavior

```python
import torch
from torch.nn.utils.rnn import pack_sequence, pad_packed_sequence

seqs = [torch.randn(i, 1) for i in range(10, 0, -1)]
seqs = pack_sequence(seqs, enforce_sorted=False)
seqs = seqs.to('cuda')
print(seqs.sorted_indices.device)

try:
    pad_packed_sequence(seqs, True, 0)
except Exception as e:
    print(e)
```

outputs

```
cuda:0
```

## Environment

PyTorch version: 1.1.0
Is debug build: No
CUDA used to build PyTorch: 10.0.130

OS: Ubuntu 18.04.2 LTS
GCC version: (Ubuntu 7.4.0-1ubuntu1~18.04.1) 7.4.0
CMake version: version 3.10.2

Python version: 3.6
Is CUDA available: Yes
CUDA runtime version: Could not collect
GPU models and configuration:
GPU 0: Tesla M60
GPU 1: Tesla M60
GPU 2: GeForce RTX 2070

Nvidia driver version: 430.26
cuDNN version: Could not collect

Versions of relevant libraries:
[pip3] numpy==1.16.4
[pip3] torch==1.1.0
[pip3] torchsummary==1.5.1
[pip3] torchvision==0.3.0
[conda] Could not collect

cc @ezyang @gchanan @zou3519 @jerryzh168

**Comment by unendin** [comment]
same issue holds for unsorted_indices of course

**Comment by IDDT** [comment]
Issue is still present on `pytorch==1.3.1`

To fix, replace the following:
`X = X.to(device)`
With this:
`X = X.to(device=device)`
Provided that "X" is a packed sequence.

**Comment by zou3519** [comment]
I'm bumping the priority on this due to the number of likes. Should not be a difficult fix.

**Comment by zou3519** [comment]
This is not that easy to fix. The problem with calling `to('cuda')` is that `PackedSequence.to` does not know what 'cuda' is: it could be a device, dtype, layout, etc.

`Tensor.to` does not have this problem because it has overloads in PythonArgParser to handle this special behavior, but `PackedSequence.to` is implemented only in python and does not have access to that logic.

We could either:
- put in a special case inside `PackedSequence.to` to check if 'cuda' or 'cpu' has been passed in
- Write a python API for PythonArgParser and query the API it with the args and kwargs to determine which overload is being called and to grab arguments by name.

The first option is the easiest thing to do right now but it is not very robust if we ever decide to change the behavior of `Tensor.to`.

**Comment by peterbell10** [comment]
@mruberry has already fixed this in #27245.
